diff --git a/crypto/CMakeLists.txt b/crypto/CMakeLists.txt
--- a/crypto/CMakeLists.txt
+++ b/crypto/CMakeLists.txt
@@ -266,23 +266,18 @@ set(
 	bf/blowfish.c
 	bio/b_dump.c
 	bio/b_print.c
-	bio/b_sock.c
 	bio/bf_buff.c
 	bio/bf_nbio.c
 	bio/bf_null.c
 	bio/bio_cb.c
 	bio/bio_err.c
 	bio/bio_lib.c
 	bio/bio_meth.c
-	bio/bss_acpt.c
 	bio/bss_bio.c
-	bio/bss_conn.c
-	bio/bss_dgram.c
 	bio/bss_fd.c
 	bio/bss_file.c
 	bio/bss_mem.c
 	bio/bss_null.c
-	bio/bss_sock.c
 	bn/bn_add.c
 	bn/bn_add_sub.c
 	bn/bn_bpsw.c
@@ -619 +614,6 @@ if(UNIX)
 endif()
+if(CMAKE_SYSTEM_NAME STREQUAL "WASI")
+	set(CRYPTO_SRC ${CRYPTO_SRC} compat/crypto_lock_wasi.c)
+	set(CRYPTO_SRC ${CRYPTO_SRC} ui/ui_openssl_wasi.c)
+endif()
+
diff --git a/crypto/compat/crypto_lock_wasi.c b/crypto/compat/crypto_lock_wasi.c
new file mode 100644
--- /dev/null
+++ b/crypto/compat/crypto_lock_wasi.c
@@ -0,0 +1,17 @@
+#include <openssl/crypto.h>
+
+#include "crypto_internal.h"
+
+void
+CRYPTO_lock(int mode, int type, const char *file, int line)
+{
+}
+LCRYPTO_ALIAS(CRYPTO_lock);
+
+int
+CRYPTO_add_lock(int *pointer, int amount, int type, const char *file,
+    int line)
+{
+	return (*pointer += amount);
+}
+LCRYPTO_ALIAS(CRYPTO_add_lock);
diff --git a/crypto/ui/ui_openssl_wasi.c b/crypto/ui/ui_openssl_wasi.c
new file mode 100644
--- /dev/null
+++ b/crypto/ui/ui_openssl_wasi.c
@@ -0,0 +1,10 @@
+#include <openssl/ui.h>
+
+#include "ui_local.h"
+
+const UI_METHOD *
+UI_OpenSSL(void)
+{
+	return UI_null();
+}
+LCRYPTO_ALIAS(UI_OpenSSL);
